## 62% of end-end tests have passed
```
Total executed tests: 8
Failed tests: 3
Expected to fail tests: 0
Passed tests: 5
```
```
IDs of failed tests: 727230,727229,703133 
```
<h3>Failed tests (3)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=fix/select-network-to-receive-subtitle&PR_ID=21872&APK_NAME=21872.apk&TR_CASE_IDS=727230,727229,703133">Rerun failed tests</a></li><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3757850">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div># STEP: Check that removed user is not shown in the list anymore</div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:240: in test_restore_multiaccount_with_waku_backup_remove_profile_switch
    self.errors.verify_no_errors()
base_test_case.py:209: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 zQ3...dWXh5 was not restored as a contact from waku backup!
E    zQ3...Vacac was not restored as a contact from waku backup!
E    admin_open was not restored from waku-backup!!
E    member_open was not restored from waku-backup!!
E    admin_closed was not restored from waku-backup!!
E    member_closed was not restored from waku-backup!!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESd61Jrd6at0lbVvlS">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3757852">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td><p><blockquote><div>Device 1: Could not reach wallet view by pressing system back button </div><div>Balance is Invalid API Key (#err2)|ARBTESTNET Gwei</div></blockquote></p>

```
critical/test_wallet.py:163: in test_wallet_send_asset_from_drawer
    sender_balance, receiver_balance, eth_amount_sender, eth_amount_receiver = self._get_balances_before_tx()
critical/test_wallet.py:38: in _get_balances_before_tx
    sender_balance = self.network_api.get_balance(self.sender['wallet_address'])
../support/api/network_api.py:55: in get_balance
    return int(balance) / 1000000000000000000
 invalid literal for int() with base 10: 'Invalid API Key (#err2)|ARBTESTNET'
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3757851">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td><p><blockquote><div>Device 2: Tap on found: Button </div><div>Balance is Invalid API Key (#err2)|ARBTESTNET Gwei</div></blockquote></p>

```
critical/test_wallet.py:129: in test_wallet_send_eth
    sender_balance, receiver_balance, eth_amount_sender, eth_amount_receiver = self._get_balances_before_tx()
critical/test_wallet.py:38: in _get_balances_before_tx
    sender_balance = self.network_api.get_balance(self.sender['wallet_address'])
../support/api/network_api.py:55: in get_balance
    return int(balance) / 1000000000000000000
 invalid literal for int() with base 10: 'Invalid API Key (#err2)|ARBTESTNET'
```

<br/><br/></td></tr></tbody></table></details><h3>Passed tests (5)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3757849">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESPE9Pr3PdPBUXfjDA">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESNkdi9hEPiS7S7hG2">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3757853">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3757854">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3757847">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESd61Jrd6at0lbVvlS">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3757848">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESUZDT3m2tTTIxWL5o">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESPKQiEiToKpBFFqki">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/status_im/contexts/wallet/bridge/bridge_to/style.cljs
+++ b/src/status_im/contexts/wallet/bridge/bridge_to/style.cljs
@@ -3,3 +3,11 @@
 (def content-container
   {:padding-horizontal 8})
 
+(def title-container
+  {:margin-horizontal 4})
+
+(def subtitle-container
+  {:flex-direction    :row
+   :margin-bottom     12
+   :margin-horizontal 20
+   :align-items       :center})

--- a/src/status_im/contexts/wallet/bridge/bridge_to/view.cljs
+++ b/src/status_im/contexts/wallet/bridge/bridge_to/view.cljs
@@ -4,45 +4,38 @@
     [quo.foundations.resources :as quo.resources]
     [quo.theme]
     [react-native.core :as rn]
+    [status-im.constants :as constants]
     [status-im.contexts.wallet.bridge.bridge-to.style :as style]
     [status-im.contexts.wallet.common.account-switcher.view :as account-switcher]
     [status-im.contexts.wallet.common.utils :as utils]
-    [status-im.contexts.wallet.common.utils.networks :as network-utils]
     [status-im.setup.hot-reload :as hot-reload]
     [utils.i18n :as i18n]
     [utils.re-frame :as rf]))
 
 (defn- bridge-token-component
   []
-  (fn [{:keys [chain-id network-name]} {:keys [supported-networks] :as token}]
-    (let [network                               (rf/sub [:wallet/network-details-by-chain-id chain-id])
-          currency                              (rf/sub [:profile/currency])
-          currency-symbol                       (rf/sub [:profile/currency-symbol])
-          prices-per-token                      (rf/sub [:wallet/prices-per-token])
-          selected-network                      (rf/sub [:wallet/send-network])
-          {selected-network-chain-id :chain-id} selected-network
-          balance                               (utils/calculate-total-token-balance token [chain-id])
-          crypto-value                          (utils/get-standard-crypto-format token
-                                                                                  balance
-                                                                                  prices-per-token)
-          fiat-value                            (utils/calculate-token-fiat-value
-                                                 {:currency         currency
-                                                  :balance          balance
-                                                  :token            token
-                                                  :prices-per-token prices-per-token})
-          fiat-formatted                        (utils/fiat-formatted-for-ui currency-symbol
-                                                                             fiat-value)
-          token-available-on-network?           (network-utils/token-available-on-network?
-                                                 supported-networks
-                                                 chain-id)]
+  (fn [{:keys [chain-id network-name]} token]
+    (let [network          (rf/sub [:wallet/network-details-by-chain-id chain-id])
+          currency         (rf/sub [:profile/currency])
+          currency-symbol  (rf/sub [:profile/currency-symbol])
+          prices-per-token (rf/sub [:wallet/prices-per-token])
+          token-symbol     (rf/sub [:wallet/wallet-send-token-symbol])
+          balance          (utils/calculate-total-token-balance token [chain-id])
+          crypto-value     (utils/get-standard-crypto-format token
+                                                             balance
+                                                             prices-per-token)
+          fiat-value       (utils/calculate-token-fiat-value
+                            {:currency         currency
+                             :balance          balance
+                             :token            token
+                             :prices-per-token prices-per-token})
+          fiat-formatted   (utils/fiat-formatted-for-ui currency-symbol
+                                                        fiat-value)]
       [quo/network-list
        {:label         (name network-name)
         :network-image (quo.resources/get-network (:network-name network))
-        :token-value   (str crypto-value " " (:symbol token))
+        :token-value   (str crypto-value " " token-symbol)
         :fiat-value    fiat-formatted
-        :state         (if (and token-available-on-network? (not= chain-id selected-network-chain-id))
-                         :default
-                         :disabled)
         :on-press      #(rf/dispatch [:wallet/select-bridge-network
                                       {:network-chain-id chain-id
                                        :stack-id         :screen/wallet.bridge-to}])}])))
@@ -52,29 +45,49 @@
   (let [network-details  (rf/sub [:wallet/network-details])
         account          (rf/sub [:wallet/current-viewing-account])
         token            (rf/sub [:wallet/wallet-send-token])
+        network          (rf/sub [:wallet/send-network])
+        network-name     (:full-name network)
         token-symbol     (:symbol token)
         tokens           (:tokens account)
         mainnet          (first network-details)
-        layer-2-networks (rest network-details)
+        layer-2-networks (filter #(not= (:chain-id %) (:chain-id network))
+                                 (rest network-details))
         account-token    (some #(when (= token-symbol (:symbol %)) %) tokens)
         account-token    (when account-token
                            (assoc account-token
                                   :networks           (:networks token)
                                   :supported-networks (:supported-networks token)))
         bridge-to-title  (i18n/label :t/select-network-to-receive)]
-
     (hot-reload/use-safe-unmount #(rf/dispatch [:wallet/clean-bridge-to-selection]))
-
     [rn/view
      [account-switcher/view
       {:on-press            #(rf/dispatch [:navigate-back])
        :icon-name           :i/arrow-left
        :accessibility-label :top-bar
        :switcher-type       :select-account}]
-     [quo/page-top {:title bridge-to-title}]
-     [rn/view style/content-container
-      [bridge-token-component (assoc mainnet :network-name :t/mainnet) account-token]]
-
+     [quo/page-top
+      {:title           bridge-to-title
+       :container-style {:padding-vertical 0}}]
+     [rn/view {:style style/subtitle-container}
+      [quo/context-tag
+       {:token token-symbol
+        :label token-symbol
+        :size  24
+        :type  :token}]
+      [quo/text
+       {:size                :paragraph-2
+        :weight              :medium
+        :style               style/title-container
+        :accessibility-label :send-label}
+       (i18n/label :t/from)]
+      [quo/context-tag
+       {:network-name network-name
+        :type         :network
+        :network-logo (:source network)
+        :size         24}]]
+     (when (not= (:chain-id network) constants/ethereum-mainnet-chain-id)
+       [rn/view style/content-container
+        [bridge-token-component (assoc mainnet :network-name :t/mainnet) account-token]])
      [quo/divider-label (i18n/label :t/layer-2)]
      [rn/flat-list
       {:data                    layer-2-networks

--- a/src/status_im/contexts/wallet/events.cljs
+++ b/src/status_im/contexts/wallet/events.cljs
@@ -408,7 +408,8 @@
             ;; If the token has a balance in only one account (or this was dispatched from the
             ;; account screen) and the network is already set, navigate forward in the bridge flow.
             (some? network)
-            [[:dispatch
+            [(when to-address [:dispatch [:wallet/switch-current-viewing-account to-address]])
+             [:dispatch
               [:wallet/wizard-navigate-forward
                {:current-screen stack-id
                 :start-flow?    start-flow?

